### PR TITLE
Coerce object at runtime

### PR DESCRIPTION
**Pull request by Neu2o**

### CODE REVIEW

File: src/ax3/filters/CoerceToBool.hx [hunk 1]
```diff
@@ -27,7 +27,7 @@ class CoerceToBool extends AbstractFilter {
 			case TTBoolean:
 				e; // shouldn't happen really
 
-			case TTFunction | TTFun(_) | TTClass | TTObject(_) | TTInst(_) | TTStatic(_) | TTArray(_) | TTVector(_) | TTRegExp | TTXML | TTXMLList | TTDictionary(_, _):
```
Comment (nadako) [inline src/ax3/filters/CoerceToBool.hx:30]: Why tho? Is this about empty strings typed as objects?

File: src/ax3/filters/ToString.hx [hunk 2]
```diff
@@ -22,7 +22,7 @@ class ToString extends AbstractFilter {
 					case [TTString, TTString]:
 						e; // ok
 
-					case [TTAny, TTString]:
+					case [TTAny | TTObject(_), TTString]:
```
Comment (nadako) [inline src/ax3/filters/ToString.hx:25]: So before it reported an error?

File: src/ax3/filters/CoerceToBool.hx (hunk 1 above)
Comment (Neu2o) [reply src/ax3/filters/CoerceToBool.hx:30]: This here is about having a boolean stored in an Object and coercing it back to boolean. It's a bit unfortunate again as we have some abstract data storage class, that is later used with the proper type, in this case boolean. Similar to this:

```
class A {
    protected var _data:Object;
}

class B extends A {
    public function getData():boolean {
        return _data;
    }
    public function setData(value:boolean) {
        _data = value;
    }
}
```

It converted to:

```
public function getData():Bool {
    return _data != null;
}
```

but that's not correct as AS3 will coerce to boolean at run-time, similar to what we already have for ASAny. Now ASAny and ASObject behave the same when coercing to Bool.

File: src/ax3/filters/ToString.hx (hunk 2 above)
Comment (Neu2o) [reply src/ax3/filters/ToString.hx:25]: No, it took the `case _:` case; so nothing really changed here. I just moved it explicitly into the ASAny case to also take advantage from the "handle at run-time" comment. ASAny and ASObject behave the same in this regard, so I put them together.

File: src/ax3/filters/CoerceToBool.hx (hunk 1 above)
Comment (nadako) [reply src/ax3/filters/CoerceToBool.hx:30]: Right. Sigh :)

**Pull request merged**